src/state/trajectory/logics.ts

```diff
@@ -345,6 +353,102 @@ const loadFileViaUrl = createLogic({
     type: LOAD_FILE_VIA_URL,
 });
 
+// configures the controller for file conversion and sends server health checks
+const configureFileConversionLogic = createLogic({
+    process(
+        deps: ReduxLogicDeps,
+        dispatch: <T extends AnyAction>(action: T) => T,
+        done
+    ) {
+        const { getState } = deps;
+
+        // TODO: Most likely this will eventually replace the config up top
+        // but for development purposes it's here
+        // until we switch to Octopus, make sure it matches your local instance.
+        const netConnectionConfig: NetConnectionParams = {
+            serverIp: "0.0.0.0",
+            serverPort: 8765,
+            useOctopus: true,
+            secureConnection: false,
+        };
+        // check if a controller exists and has the right configuration
+        let controller = getSimulariumController(getState());
+        if (!controller || !controller.remoteWebsocketClient) {
```

Comment:
do you need to create a new controller to set the network settings? If the controller exists can't you just update it?

Reply:
yes, i think we could just update it if it exists. i can add that conditional.

Reply:
```
   if (!controller) {
            dispatch(
                setConversionStatus({
                    status: CONVERSION_NO_SERVER,
                })
            );
            controller = new SimulariumController({
                netConnectionSettings: netConnectionConfig,
            });
            dispatch(setSimulariumController(controller));
        } 
        else if (!controller.remoteWebsocketClient) {
            controller.configureNetwork(netConnectionConfig);
        }